cscs-checks/apps/vasp/vasp_check.py

```diff
@@ -7,130 +7,124 @@
-                    'OMP_NUM_THREADS': str(self.num_cpus_per_task),
-                    'OMP_PLACES': 'cores',
-                    'OMP_PROC_BIND': 'close'
-                }
-        else:
-            if self.current_system.name in ['daint']:
-                self.num_tasks = 32
-                self.num_tasks_per_node = 2
-                self.num_cpus_per_task = 18
-
-    @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:mc': {'time': (65.0, None, 0.15, 's')},
-                'daint:mc': {'time': (65.0, None, 0.15, 's')},
-                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
-                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
-            },
-            'large': {
-                'daint:mc': {'time': (55.0, None, 0.15, 's')},
-                'eiger:mc': {'time': (100.0, None, 0.10, 's')},
-                'pilatus:mc': {'time': (100.0, None, 0.10, 's')}
-            }
+    def setup_system_filtering(self):
+        self.descr += f' ({self.num_nodes} node(s))'
+
+        # setup system filter
+        valid_systems = {
+            6: ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                'eiger:mc', 'pilatus:mc'],
+            16: ['daint:gpu', 'daint:mc', 'eiger:mc']
         }
-        self.reference = references[self.scale]
 
-    @run_before('run')
-    def set_task_distribution(self):
-        self.job.options = ['--distribution=block:block']
-
-    @run_before('run')
-    def set_cpu_binding(self):
-        self.job.launcher.options = ['--cpu-bind=cores']
+        try:
+            self.valid_systems = valid_systems[self.num_nodes]
+        except KeyError:
+            self.valid_systems = []
 
+        # setup programming environment filter
+        if self.current_system.name in ['eiger', 'pilatus']:
+            self.valid_prog_environs = ['cpeIntel']
+        else:
+            self.valid_prog_environs = ['builtin']
 
-@rfm.simple_test
-class VASPGpuCheck(VASPCheck):
-    valid_systems = ['daint:gpu']
-    num_gpus_per_node = 1
-    num_tasks_per_node = 1
-    num_cpus_per_task = 12
-    use_multithreading = False
 
-    @run_after('init')
-    def setup_test(self):
-        self.descr = (f'VASP GPU check (version: {self.scale}, '
-                      f'{self.variant})')
-        if self.scale == 'small':
-            self.valid_systems += ['dom:gpu']
-            self.num_tasks = 6
-        else:
-            self.num_tasks = 16
-
-    @run_before('performance')
-    def set_reference(self):
-        references = {
-            'small': {
-                'dom:gpu': {'time': (65.0, None, 0.15, 's')},
-                'daint:gpu': {'time': (65.0, None, 0.15, 's')}
-            },
-            'large': {
-                'daint:gpu': {'time': (55.0, None, 0.15, 's')}
+    @run_before('run')
+    def setup_run(self):
+        # set auto-detected architecture
+        proc = self.current_partition.processor
```

Comment:
You need to add a call to `skip_if_no_procinfo()` here, so as to skip the test is the processor topology info is not available.

Reply:
Added the call to `self.skip_if_no_procinfo()` as suggested